src/axom/sidre/core/View.cpp

```diff
@@ -295,6 +295,44 @@ View* View::reallocate(const DataType& dtype)
   return this;
 }
 
+/*
+ *************************************************************************
+ *
+ * Reshape an array View.
+ *
+ *************************************************************************
+ */
+View* View::reshapeArray(int ndims, const IndexType* shape)
+{
+  SLIC_ERROR_IF(m_state != BUFFER && m_state != EXTERNAL,
```

Comment:
Implementation choice:  The usual error-checking practice in Sidre for these cases where a required precondition is not met is to use SLIC_CHECK_MSG for a warning and return as a no-op, so that calling code has a chance to check the results and recover.

Reply:
Thanks @nselliott  Good idea to be consistent.  Fixed.